drivers/pinctrl/pinctrl_ite_it8xxx2.c

```diff
@@ -34,13 +34,42 @@ struct pinctrl_it8xxx2_config {
 	uint8_t volt_sel_mask[GPIO_GROUP_MEMBERS];
 };
 
+struct pinctrl_it8xxx2_ksi_kso {
+	/*
+	 * KSI[7:0]/KSO[15:8]/KSO[7:0] port gpio control register
+	 * (bit mapping to pin)
+	 */
+	uint8_t *reg_gctrl;
+	/* KSI[7:0]/KSO[15:8]/KSO[7:0] port control register */
+	uint8_t *reg_ctrl;
+	/*
+	 * KSO push-pull/open-drain bit of KSO[15:0] control register
+	 * (this bit apply to all pins)
+	 */
+	int pp_od_mask;
+	/*
+	 * KSI/KSO pullup bit of KSI[7:0]/KSO[15:0] control register
+	 * (this bit apply to all pins)
+	 */
+	int pullup_mask;
+};
+
+struct pinctrl_it8xxx2_config {
+	bool gpio_group;
+	union config {
+		struct pinctrl_it8xxx2_gpio gpio;
+		struct pinctrl_it8xxx2_ksi_kso ksi_kso;
+	} config;
+};
+
 static int pinctrl_it8xxx2_set(const pinctrl_soc_pin_t *pins)
 {
 	const struct pinctrl_it8xxx2_config *pinctrl_config = pins->pinctrls->config;
+	const struct pinctrl_it8xxx2_gpio *gpio = &(pinctrl_config->config.gpio);
```

Comment:
If you drop the `config` label on the union, this becomes:
```suggestion
	const struct pinctrl_it8xxx2_gpio *gpio = &(pinctrl_config->gpio);
```

Reply:
Done